bot/cogs/help.py

```diff
@@ -70,499 +71,304 @@ class HelpSession:
-            paginator.add_line(f'**{self.query.name}**')
-
-        if self.description:
-            paginator.add_line(f'*{self.description}*')
-
-        # list all children commands of the queried object
-        if isinstance(self.query, (commands.GroupMixin, Cog)):
-
-            # remove hidden commands if session is not wanting hiddens
-            if not self._show_hidden:
-                filtered = [c for c in self.query.commands if not c.hidden]
-            else:
-                filtered = self.query.commands
-
-            # if after filter there are no commands, finish up
-            if not filtered:
-                self._pages = paginator.pages
-                return
-
-            # set category to Commands if cog
-            if isinstance(self.query, Cog):
-                grouped = (('**Commands:**', self.query.commands),)
-
-            # set category to Subcommands if command
-            elif isinstance(self.query, commands.Command):
-                grouped = (('**Subcommands:**', self.query.commands),)
-
-                # don't show prefix for subcommands
-                prefix = ''
-
-            # otherwise sort and organise all commands into categories
-            else:
-                cat_sort = sorted(filtered, key=self._category_key)
-                grouped = itertools.groupby(cat_sort, key=self._category_key)
-
-            # process each category
-            for category, cmds in grouped:
-                cmds = sorted(cmds, key=lambda c: c.name)
-
-                # if there are no commands, skip category
-                if len(cmds) == 0:
-                    continue
-
-                cat_cmds = []
-
-                # format details for each child command
-                for command in cmds:
+    async def command_formatting(self, command: Command) -> Embed:
+        """
+        Takes a command and turns it into an embed.
 
-                    # skip if hidden and hide if session is set to
-                    if command.hidden and not self._show_hidden:
-                        continue
+        It will add an author, command signature + help, aliases and a note if the user can't run the command.
+        """
+        embed = Embed()
+        embed.set_author(name="Command Help", icon_url=constants.Icons.questionmark)
 
-                    # see if the user can run the command
-                    strikeout = ''
+        parent = command.full_parent_name
 
-                    # Patch to make the !help command work outside of #bot-commands again
-                    # This probably needs a proper rewrite, but this will make it work in
-                    # the mean time.
-                    try:
-                        can_run = await command.can_run(self._ctx)
-                    except CheckFailure:
-                        can_run = False
+        name = str(command) if not parent else f"{parent} {command.name}"
+        fmt = f"**```{PREFIX}{name} {command.signature}```**\n"
```

Comment:
I also hate `fmt` as a variable name.

can you also just go through this file and remove other instances of [ninja code](https://javascript.info/ninja-code)?